src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,331 @@
+    };
+  }, [currentPage, shownPerPage]);
+
+  return (
+    <div className="pagination">
+      <div className="pagination__item--prev">
+        {isDisabledLeft ? (
+          <span className="pagination__link disabled">{'<'}</span>
+        ) : (
+          <SearchLink
+            className="pagination__link"
+            data-cy="paginationLeft"
+            params={{ [SearchParams.PageNumber]: (currentPage - 1).toString() }}
+            onClick={() => onSetCurrentPage(currentPage - 1)}
+          >
+            {'<'}
+          </SearchLink>
+        )}
+      </div>
+
+      <ul className="pagination__list" data-cy="pagination">
+        {!isLeftPoints && !isRightsPoints && allNumbersPages.map(pageNumber => (
+          <li
+            key={`page-number-${pageNumber}`}
+            className="pagination__item"
+          >
+            <SearchLink
+              className={classNames(
+                'pagination__link', { active: currentPage === pageNumber },
+              )}
+              params={{ [SearchParams.PageNumber]: pageNumber.toString() }}
+              onClick={() => onSetCurrentPage(pageNumber)}
+            >
+              {pageNumber}
+            </SearchLink>
+          </li>
+        ))}
+
+        {isLeftPoints && isRightsPoints && (
+          <>
+            <li className="pagination__item">
+              <SearchLink
+                className={classNames(
+                  'pagination__link',
+                  { active: currentPage === allNumbersPages[0] },
+                )}
+                params={{
+                  [SearchParams.PageNumber]: allNumbersPages[0].toString(),
+                }}
+                onClick={() => onSetCurrentPage(allNumbersPages[0])}
+              >
+                {allNumbersPages[0]}
+              </SearchLink>
+            </li>
+
+            <li className="pagination__item">
+              <span className="pagination__points">...</span>
+            </li>
+
+            {allNumbersPages.filter(pageNum => {
+              const leftAndRight = Math.floor((pageTogether - 1) / 2);
+
+              let lastRight = pageTogether % 2 === 0 ? 1 : 0;
+              const lastLeft = currentPage - leftAndRight;
+
+              lastRight = lastRight + currentPage + leftAndRight;
+
+              return pageNum >= lastLeft && pageNum <= lastRight;
+            }).map(pageNumber => (
+              <li
+                key={`page-number-${pageNumber}`}
+                className="pagination__item"
+              >
+                <SearchLink
+                  className={classNames(
+                    'pagination__link', { active: currentPage === pageNumber },
+                  )}
+                  params={{ [SearchParams.PageNumber]: pageNumber.toString() }}
+                  onClick={() => onSetCurrentPage(pageNumber)}
+                >
+                  {pageNumber}
+                </SearchLink>
+              </li>
+            ))}
```

Comment:
Consider moving this logic outside of jsx
```suggestion
            {allNumbersPagesLayout}
```